src/metrics/misc.ts

```diff
@@ -242,6 +252,55 @@ export class MetadataMetric implements Metric {
       };
     }
 
+
+    // identify game endings
+    let gameEnd = metadata.find(item => item.startsWith('[Termination'));
+    const lastMove = game[game.length - 1].move;
+
+    if (gameEnd === '[Termination "Normal"]') {
+      if (lastMove.originalString.includes('#') || this.chess.isCheckmate()) {
+        this.gameEndings['checkmate'] = (this.gameEndings['checkmate'] || 0) + 1;
+      } else if (result === '[Result "1/2-1/2"]') {
+        this.gameEndings['draw'] = (this.gameEndings['draw'] || 0) + 1;
+      } else if (this.chess.isStalemate()) {
+        this.gameEndings['stalemate'] = (this.gameEndings['stalemate'] || 0) + 1;
+      } else if (this.chess.isInsufficientMaterial()) {
+        this.gameEndings['insufficient material'] = (this.gameEndings['insufficient material'] || 0) + 1;
+      } else {
+        this.gameEndings['resignation'] = (this.gameEndings['resignation'] || 0) + 1;
+      }
+    } else if (gameEnd === '[Termination "Time forfeit"]') {
+      this.gameEndings['time out'] = (this.gameEndings['time out'] || 0) + 1;
+    } 
+
+    // check for threefold repetition (currently not checking if remaining castling rights and the possibility to capture en passant are the same per https://en.wikipedia.org/wiki/Threefold_repetition)
+    // check for 50-moves rule
+    let fiftyMovesCount = 0;
+    let boardPositions:  { [board: string]: number } =  {};
+    let threefoldRepetitionFound = false; // check if threeFoldRepetition has been found
+    const lastBoardString = JSON.stringify(game[game.length - 1].board) + this.chess.turn();
```

Comment:
this doesn't capture castling state, or EP

Reply:
yes look at comments line 276. im starting to think this might be the reason for why multiple threefold reps are being identified, but im not 100% sure and it seems unlikely